clang/include/clang/Sema/Sema.h

```diff
@@ -13055,8 +13103,10 @@ class Sema final {
   /// if (!S.Context.getTargetInfo().hasFloat128Type() &&
   ///     S.getLangOpts().SYCLIsDevice)
   ///   SYCLDiagIfDeviceCode(Loc, diag::err_type_unsupported) << "__float128";
-  SemaDiagnosticBuilder SYCLDiagIfDeviceCode(SourceLocation Loc,
-                                             unsigned DiagID);
+  SemaDiagnosticBuilder SYCLDiagIfDeviceCode(
+      SourceLocation Loc, unsigned DiagID,
+      DeviceDiagnosticReason Reason =
+          static_cast<DeviceDiagnosticReason>(DDR_SYCL | DDR_ESIMD));
```

Comment:
If we use the `BitmaskEnum` machinery, this `static_cast` can be removed.

Reply:
Done